lib/vdc/ArbitrarilyOrientedRegularGrid.cpp

```diff
@@ -224,3 +224,31 @@ void ArbitrarilyOrientedRegularGrid::_populateData(
 }
 
 // clang-format on
+
+glm::vec3 ArbitrarilyOrientedRegularGrid::GetOffsetNormal(const planeDescription &pd)
+{
+    glm::vec3 dataMin(pd.boxMin[0], pd.boxMin[1], pd.boxMin[2]);
+    glm::vec3 dataMax(pd.boxMax[0], pd.boxMax[1], pd.boxMax[2]);
+    glm::vec3 o(pd.origin[0], pd.origin[1], pd.origin[2]);
+    glm::vec3 r(pd.rotation[0], pd.rotation[1], pd.rotation[2]);
+    r = r * (float)M_PI / 180.f;
+    auto q = glm::quat(r);
+    auto n = q * glm::vec3(0, 0, 1);
+    return n;
+}
+
+std::pair<float, float> ArbitrarilyOrientedRegularGrid::GetOffsetRange(const planeDescription &pd)
+{
+    auto      n = GetOffsetNormal(pd);
+    glm::vec3 dataMin(pd.boxMin[0], pd.boxMin[1], pd.boxMin[2]);
+    glm::vec3 dataMax(pd.boxMax[0], pd.boxMax[1], pd.boxMax[2]);
+    glm::vec3 o(pd.origin[0], pd.origin[1], pd.origin[2]);
+    float     t0, t1;
+    glm::vec3 tMin = (dataMin - o) / n;
+    glm::vec3 tMax = (dataMax - o) / n;
+    glm::vec3 bt1 = min(tMin, tMax);
+    glm::vec3 bt2 = max(tMin, tMax);
+    t0 = max(max(bt1.x, bt1.y), bt1.z);
+    t1 = min(min(bt2.x, bt2.y), bt2.z);
+    return std::pair<float, float>(t0, t1);
+}
```

Comment:
I think there may be a miscalculation for the range and offset.  

Case 1)
1) Load a katrina file, then create and enable a Contour renderer
2) Rotate by 20 degrees on X
3) Set the offset to its maximum value.  The contour will just barely move away from its original location.

Case 2)
With the dataset below, do the following:
1) Create and enable a contour renderer
2) Rotate it by 20 degrees on X
3) Move the offset to its minimum or maximum value.  At these values the contour should only intersect at an edge of the domain, which would render nothing.  Note in the screenshot below that the green rectangle is well within the domain, not intersecting the edge.
[256i.zip](https://github.com/NCAR/VAPOR/files/8017528/256i.zip)
![Screen Shot 2022-02-07 at 10 41 51 AM](https://user-images.githubusercontent.com/9522770/152843934-f9f84ec5-d697-43c6-ad4a-7db29ca5b45a.png)

Reply:
Case 1 is correct. See #2989

Case 2 is correct. I think you meant 45 degrees, not 20.